aws-transfer-workflow/aws-transfer-workflow.json

```diff
@@ -47,6 +68,19 @@
       "additionalProperties": false
     },
     "InputFileLocation": {
+      "description": "Specifies the location for the file being decrypted. Only applicable for the Decrypt type of workflow steps.",
+      "type": "object",
+      "properties": {
+        "S3FileLocation": {
+          "$ref": "#/definitions/S3InputFileLocation"
+        },
+        "EfsFileLocation": {
+          "$ref": "#/definitions/EfsInputFileLocation"
+        }
+      },
+      "additionalProperties": false
+    },
+    "S3FileLocation": {
       "description": "Specifies the location for the file being copied. Only applicable for the Copy type of workflow steps.",
```

Comment:
Do we also want to mention Decrypt type here? Since this structure is also being referenced by `InputFileLocation` above

Reply:
So the tricky thing about this is that this S3FileLocation is only used for COPY, while Decrypt uses the whole InputFileLocation. There's no way around this because we don't allow EFS Copy (we throw a InvalidRequestException), and the Contract Test tests for all combination of property types (i.e. EFS Copy) and that fails.